src/split_single_dyn_dim.cpp

```diff
@@ -64,19 +60,18 @@ has_one_unique_dyn_dim(const std::unordered_map<std::string, shape>& param_shape
     for(const auto& param : dyn_params)
     {
         const auto& dds   = param.second.dyn_dims();
-        int num_non_fixed = 0;
-        for(const auto& dd : dds)
-        {
+        auto num_non_fixed = std::count_if(dds.cbegin(), dds.cend(), [&](auto dd) {
             if(not dd.is_fixed())
             {
-                num_non_fixed += 1;
-                // catch more than one non-fixed dynamic_dimension
-                if(num_non_fixed > 1)
-                {
-                    return std::nullopt;
-                }
                 ret.push_back(dynamic_dimensions_check{param.first, dd});
+                return true;
```

Comment:
Thats an interesting way to use `count_if`, not what I had originally in mind, but definitely works.